### ISSUE TITLE
Bugs in toplevel_ref_arg suggestions

### ISSUE THREAD

**Issue by CryZe** [Manishearth/rust-clippy#1100]
More Issues with rustfix here:

```rust
fn main() {
    let ref mut x: usize = 5;
    *x += 3;
    println!("{}", x);
}
```

```
src\main.rs:2:9: 2:18 warning: `ref` on an entire `let` pattern is discouraged, take a reference with `&` instead, #[warn(toplevel_ref_arg)] on by default
src\main.rs:2     let ref mut x: usize = 5;
                      ^~~~~~~~~
src\main.rs:2:5: 2:30 help: try
src\main.rs:      let x: usize = &5;
src\main.rs:2:9: 2:18 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#toplevel_ref_arg
```

It suggests an immutable reference instead of a mutable one. Also it doesn't modify the type either, so there's a type conflict too.

**Comment by mcarton** [comment]
Thanks for the report! Should be fixed in the next release.

**Comment by CryZe** [comment]
```rust
  let ref mut z = 1 + 2;
  //~^ ERROR `ref` on an entire `let` pattern is discouraged
  //~| HELP try
  //~| SUGGESTION let mut z = &(1 + 2);
```

This doesn't seem correct as `ref mut` creates a `&mut` and not a `mut &`

**Comment by mcarton** [comment]
Right, I convinced myself this was correct but experimenting on that it should be `let z = &mut (1 + 2);`
Cc @Manishearth
